pkg/registry/apis/datasource/register.go

```diff
@@ -224,7 +227,9 @@ func (b *DataSourceAPIBuilder) getPluginContext(ctx context.Context, uid string)
 	if err != nil {
 		return backend.PluginContext{}, err
 	}
-	return b.contextProvider.PluginContextForDataSource(ctx, instance)
+	pctx, err := b.contextProvider.PluginContextForDataSource(ctx, instance)
+	pctx.APIVersion = b.connectionResourceInfo.APIVersion()
```

Comment:
^^^ this is the interesting part

This tells the backend what apiversion was requested.  

NOTE: `pctx.DataSourceInstanceSettings.APIVersion` may be different! it should be the version when the datasource settings were saved